docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,910 @@
+it up again.
+
+With that scenario in mind, after VM has live migrated:
+- the default IPv4 default gateway mac will point to a node that is currently down
+- one of default IPv6 gateway paths will be pointing to a node that is currently down.
+
+To fix that for IPv4, ovn-kubernetes sends a GARP after live migration to
+reconcile the default gw mac to the new node where the VM is running [Pull Request 4964](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4964).
+
+For ipv6 there are a change to do something similar by blocking external gateway
+routers RAs [Pull Request 4852](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4852) and reconciling gateways with unsolicited router advertisements
+[Pull Request 4847](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4847).
+
+Although these fixes works, they are not very robust since messages can be lost
+or blocked so gateway do not get reconciled.
+
+This is how the topology will look after the virtual machine has being live migrated from node1 to node2
+and shutting down node1 after it.
+
+ipv4:
+```mermaid
+flowchart TD
+    classDef vmStyle fill:#0000FF,stroke:#ADD8E6,stroke-width:2px;
+    subgraph node1["node1
+    DOWN"]
+        GR-node1["GR-node1 (0a:58:64:41:00:02)"]
+    end
+    subgraph node2
+        GR-node2["GR-node2 (0a:58:64:41:00:03)"]
+        subgraph VM["Virtual Machine"]
+
+            class VM vmStyle;
+            route["default gw -> 203.203.0.1 old node1 instead of node2
+            (0a:58:64:41:00:02)"]
+        end
+        VM -->layer2-switch
+    end
+     subgraph node3
+        GR-node3["GR-node3 (0a:58:64:41:00:04)"]
+    end
+    GR-node1 <-->|"remote 100.65.0.2  (0a:58:64:41:00:02)"| layer2-switch
+    GR-node2 <-->|"router 203.203.0.1   (0a:58:64:41:00:03)"| layer2-switch
+    GR-node3 <-->|"remote 100.65.0.4   (0a:58:64:41:00:04)"| layer2-switch
+
+    class VM vmStyle;
+```
+
+ipv6:
+```mermaid
+flowchart TD
+    classDef vmStyle fill:#0000FF,stroke:#ADD8E6,stroke-width:2px;
+
+    subgraph node1["node1 (DOWN)"]
+
+        GR-node1["GR-node1 (fe80::858:64ff:fe41:2)"]
+    end
+    subgraph node2
+        GR-node2["GR-node2 (fe80::858:64ff:fe41:3)"]
+        layer2-switch["Layer2 Switch"]
+        subgraph VM["Virtual Machine"]
+
+            class VM vmStyle;
+            route["default gw
+            down(fe80::858:64ff:fe41:2)
+            fe80::858:64ff:fe41:3
+            fe80::858:64ff:fe41:4"]
+        end
+        layer2-switch--> VM
+    end
+    subgraph node3
+        GR-node3["GR-node3 (fe80::858:64ff:fe41:4)"]
+    end
+
+    GR-node2 --> layer2-switch
+    GR-node3 --> layer2-switch
+
+    class VM vmStyle;
+```
+
+### Layer2 topology limitations for EIP
+
+The current layer2 topology has some limitations with Egress IP.
+When multiple IPs are assigned to an Egress IP, and a pod is local to one
+of the egress nodes, only the egress path local to that node will be used.
```

Comment:
we still have multiple hops for a LRP with reroute action so we do still have LB within the cluster but as you said earlier, we only support one GW as one of the next hops.